### PR TITLE
utils: remove DSSInternalDict

### PR DESCRIPTION
**Pull request by jfyuen**
This PR replaces `DSSInternalDict` (which did not have a lot of features) to use an `_internal_dict` object in each subclass.

### CODE REVIEW

**Comment by nicolasservel** [comment]
Is there any particular rationale ? Because this PR does reintroduce some boiler plate code that was factored in DSSInternalDict

**Comment by instanceofme** [comment]
Yes, the advantages we initially found to `DSSInternalDict` were actually negated:
- The `get` method is not that useful and looks a bit weird
- The `get_raw` would be useful but factorizes the doc whereas we'd like to customize it for each class so we'll have to override anyway
- The `__repr__` exposes internal content whereas we might not want to actively push this when printing the object

All things considered, not that useful ^^

File: dataikuapi/dss/ml.py [hunk 1]
```diff
@@ -824,28 +823,42 @@ def get_partial_dependencies(self):
...
+    def get_raw(self):
+        """
+        Gets the raw dictionary of the actual data
+
+        :rtype: dict
+        """
+        return self._internal_dict
+
+    def __repr__(self):
+        return self.__class__.__name__ + "(" + self._internal_dict.__repr__() + ")"
```
Comment (instanceofme) [inline dataikuapi/dss/ml.py]: Not sure this is useful

File: dataikuapi/dss/ml.py [hunk 2]
```diff
@@ -862,22 +875,36 @@ def get_prediction_info(self):
...
```
Comment (instanceofme) [inline dataikuapi/dss/ml.py]: Should return the modality (category value or numerical bin) rather than the full data

File: dataikuapi/dss/ml.py [hunk 3]
```diff
@@ -985,26 +1012,37 @@ def __repr__(self):
...
```
Comment (instanceofme) [inline dataikuapi/dss/ml.py]: Ditto, can maybe indicate the column name, the type and the number of modalities

File: dataikuapi/dss/ml.py [hunk 4]
```diff
@@ -1046,88 +1084,130 @@ def get_modality_data(self, definition):
...
```
Comment (instanceofme) [inline dataikuapi/dss/ml.py]: Ditto, maybe just a list of columns for which analyses where computed?

File: dataikuapi/dss/ml.py [hunk 5]
```diff
@@ -1046,88 +1084,130 @@ def get_modality_data(self, definition):
...
```
Comment (instanceofme) [inline dataikuapi/dss/ml.py]: Ditto → column name + type?

File: dataikuapi/dss/ml.py [hunk 6]
```diff
@@ -1046,88 +1084,130 @@ def get_modality_data(self, definition):
...
```
Comment (instanceofme) [inline dataikuapi/dss/ml.py]: Ditto → column list?

File: dataikuapi/dss/ml.py (hunk 5 above)
Comment (jfyuen) [reply dataikuapi/dss/ml.py]: I couldn't find the type for this class, only the feature.

File: dataikuapi/dss/ml.py [hunk 7]
```diff
@@ -1046,88 +1093,131 @@ def get_modality_data(self, definition):
...
+        Gets the raw dictionary of subpopulation analyses
+
+        :rtype: dict
+        """
+        return self._internal_dict
+
+    def __repr__(self):
+        return "{cls}(prediction_type={type}, analyses={analyses})".format(cls=self.__class__.__name__,
+                                                                           type=self.prediction_type,
+                                                                           analyses=self.list_analyses)
```
Comment (nicolasservel) [inline dataikuapi/dss/ml.py]:
```suggestion
                                                                           analyses=self.list_analyses())
```

**Comment by jfyuen** [comment]
> I wonder if we need to initialize `_internal_dict` if `data is None` as the class will be totally unusable in that case.

I wondered the same thing. I think it all depends if we want to raise an error on instance creation or when accessing the `dict`. I would prefer to test for `None` at dict creation, but that doesn't prevent using an empty `dict` that would fail later. There is no good way to check everything at object creation...

**Comment by nicolasservel** [comment]
> > I wonder if we need to initialize `_internal_dict` if `data is None` as the class will be totally unusable in that case.
> 
> I wondered the same thing. I think it all depends if we want to raise an error on instance creation or when accessing the `dict`. I would prefer to test for `None` at dict creation, but that doesn't prevent using an empty `dict` that would fail later. There is no good way to check everything at object creation...

I agree with you, there is perfect check. But here, creating an empty dict does not bring anything IMO compared to just initialize the _internal_dict with `None`, it will just produce a different (and not more understandable) error message afterwards

**Comment by jfyuen** [comment]
> > > I wonder if we need to initialize `_internal_dict` if `data is None` as the class will be totally unusable in that case.
> > 
> > 
> > I wondered the same thing. I think it all depends if we want to raise an error on instance creation or when accessing the `dict`. I would prefer to test for `None` at dict creation, but that doesn't prevent using an empty `dict` that would fail later. There is no good way to check everything at object creation...
> 
> I agree with you, there is perfect check. But here, creating an empty dict does not bring anything IMO compared to just initialize the _internal_dict with `None`, it will just produce a different (and not more understandable) error message afterwards

Yes, we're on the same line. Done.

**Pull request merged**